File: tests/rosettacode/caesar.clj
;; Caesar cipher with shift 13
(defn caesar-encrypt [s shift]
  (apply str
    (map (fn [c]
           (cond
             (Character/isUpperCase c)
             (char (+ (int \A) (mod (+ (- (int c) (int \A)) shift) 26)))
             (Character/isLowerCase c)
             (char (+ (int \a) (mod (+ (- (int c) (int \a)) shift) 26)))
             :else c))
         s)))

(defn caesar-decrypt [s shift]
  (caesar-encrypt s (- 26 shift)))

(let [text "The quick brown fox jumps over the lazy dog"
      shift 13
      encrypted (caesar-encrypt text shift)
      decrypted (caesar-decrypt encrypted shift)]
  (println (str "Original:  " text))
  (println (str "Encrypted: " encrypted))
  (println (str "Decrypted: " decrypted)))

File: tests/java-rosettacode/Caesar.java
public class Caesar {
    public static String encrypt(String text, int shift) {
        StringBuilder sb = new StringBuilder();
        for (char c : text.toCharArray()) {
            if (Character.isUpperCase(c)) {
                sb.append((char) ('A' + (c - 'A' + shift) % 26));
            } else if (Character.isLowerCase(c)) {
                sb.append((char) ('a' + (c - 'a' + shift) % 26));
            } else {
                sb.append(c);
            }
        }
        return sb.toString();
    }

    public static String decrypt(String text, int shift) {
        return encrypt(text, 26 - shift);
    }

    public static void main(String[] args) {
        String plaintext = "The quick brown fox";
        int shift = 11;
        String encrypted = encrypt(plaintext, shift);
        String decrypted = decrypt(encrypted, shift);
        System.out.println("Plain:     " + plaintext);
        System.out.println("Encrypted: " + encrypted);
        System.out.println("Decrypted: " + decrypted);
    }
}
